### PR TITLE
What am flow

### LINKED ISSUE #75: Intro docs: WTF is "flow"
In the intro docs, under "[Getting Connected: an Echo Server](https://tubes.readthedocs.io/en/latest/tube.html#getting-connected-an-echo-server)", after the example, which looks like this:

```python
def echoFlow(flow):
    flow.fount.flowTo(flow.drain)
```

the next paragraph makes the **outrageous** claim that "In the above example, echoFlow takes two things".  **IT TAKES ONE THING**, or I'm insane (I'm going to assume the former).

I get stuck here wondering if I can't count, and then when I accept that I'm simply being misled, I'm left wondering the degree to which this evil plot to confuse me is trying to drive me nuts, which them makes me wonder if perhaps I'm already nuts, and I should have assumed the latter.

**WTF is flow?**  Is it a type I should know about?  Or some random thingo with these two attributes attached to it?  **OMG I DON'T KNOW!**  And then it says here that **the function** is a "flow".  **WHAT?!?  Why is the argument called the thing that the function is?**

Anyway, I feel like I'd be feeling a lot less stress if this function actually took two arguments:

```python
def echoFlow(fount, drain):
    fount.flowTo(drain)
```

Then you can say it takes two thing, and even call the function a flow and I'd be cool with all of that.

Peace be unto you.

### PR DESCRIPTION
**Pull request by glyph**
Fixes #75 

Hi @wsanchez let me know if this makes the docs more legible.

### CODE REVIEW

File: docs/tube.rst [hunk 1]
```diff
@@ -28,19 +28,29 @@ On a network, that means an echo server as described in :rfc:`862`.
...
+Let's look at the full example that turns ``echo`` into a real server.
 
 :download:`echoflow.py <listings/echoflow.py>`
 
 .. literalinclude:: listings/echoflow.py
 
+To *use* ``echo`` as a server, first we have to tell Tubes that it's a :api:`tubes.itube.IDrain <drain>` that wants :api:`tubes.listening.Flow <flow>`\ s.
+We do this by wrapping it in a :api:`tubes.listening.Listener <Listener>`\ .
+
+Next, we need to actually listen on a port: we do this with Twisted's `"endpoints" API <https://twistedmatrix.com/documents/current/core/howto/endpoints.html>`_ ; specifically, we use ``serverFromString`` on the string ``"stdio:"`` by default, which treats the console as an incoming connection so we can type directly into it.
```
Comment (moshez) [inline docs/tube.rst]: "and see the results on it" or something to that effect?

File: docs/tube.rst [hunk 2]
```diff
@@ -28,19 +28,29 @@ On a network, that means an echo server as described in :rfc:`862`.
...
+
+Next, we need to convert this endpoint into a :api:`tubes.itube.IFount <fount>` of :api:`tubes.listening.Flow <flows>`.
```
Comment (moshez) [inline docs/tube.rst]: What's an "Interface" of "type"? Is this interface parametrized?

File: docs/tube.rst [hunk 3]
```diff
@@ -94,6 +104,17 @@ You can test it out with ``telnet localhost 4321``.
 
 However, this example still performs no processing of the data that it is receiving.
 
+A Brief Aside About Types
```
Comment (moshez) [inline docs/tube.rst]: I think this answers my earlier question, but maybe mention this aside earlier?

File: docs/tube.rst [hunk 4]
```diff
@@ -94,6 +104,17 @@ You can test it out with ``telnet localhost 4321``.
...
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Each fount, and each drain, have a type associated with them: in the fount's case, the type of data it produces, and in the drain's case, the type of data that it accepts.
+You can inspect these using the :api:`tubes.itube.IFount.outputType <outputType>` and :api:`tubes.itube.IDrain.inputType <inputType>` attributes of founts and drains respectively.
+Even in our tiny example, we already have two types of founts: a fount of ``bytes`` — one for each connection — and a fount of :api:`tubes.listening.Flow <flow>`\ s — the listening port).
+We have a drain for ``bytes``, also on each connection, and a drain for :api:`tubes.listening.Flow <flow>`\ s: the :api:`tubes.listening.Listener <listener>` wrapped around ``echo``\ .
+
+Attempting to hook up a fount and a drain of mismatched types will usually result in an immediate ``TypeError``, which is a helpful debugging tool.
```
Comment (moshez) [inline docs/tube.rst]: "Usually" is a scary word here. What's the exception?

File: docs/tube.rst [hunk 5]
```diff
@@ -113,12 +134,12 @@ Luckily Tubes implements this for us, with the handy :api:`tubes.framing` module
...
     However, if you're designing your own network protocol, please consider using a length-prefixed framing mechanism, such as :api:`tubes.framing.netstringsToStrings <netstrings>`.
 
-Much like in the echo example, we need a flow function which sets up the flow of data from a fount to a drain.
+Much like in the echo example, we need a function which accepts a ``flow`` which sets up the flow of data from a fount to a drain on an individual connection.
 
 .. literalinclude:: listings/reversetube.py
    :pyobject: reverseFlow
 
-In this flow function, we have a new object, a *series of Tubes*, created by the :api:`tubes.tube.series <series>` function.
+In this function, we create a new type of object, a *series of Tubes*, created by the :api:`tubes.tube.series <series>` function.
```
Comment (moshez) [inline docs/tube.rst]: Man, this joke is, like, getting old fast...

File: docs/listings/echoflow.py [hunk 6]
```diff
@@ -4,14 +4,15 @@
 from twisted.internet.endpoints import serverFromString
 from twisted.internet.defer import Deferred, inlineCallbacks
 
-def echoFlow(flow):
+def echo(flow):
     flow.fount.flowTo(flow.drain)
 
 @inlineCallbacks
 def main(reactor, listenOn="stdio:"):
+    listening = Listener(echo)
```
Comment (wsanchez) [inline docs/listings/echoflow.py]: Why a verb and not `listener`?

File: docs/tube.rst [hunk 7]
```diff
@@ -28,19 +28,29 @@ On a network, that means an echo server as described in :rfc:`862`.
...
 .. literalinclude:: listings/echoflow.py
-    :pyobject: echoFlow
+    :pyobject: echo
 
-In the above example, ``echoFlow`` takes two things: a :api:`tubes.itube.IFount <fount>`, or a source of data, and a :api:`tubes.itube.IDrain <drain>` , or a place where data eventually goes.
-Such a function is called a "flow", because it establishes a flow of data from one place to another.
-Most often, the arguments to such a function are the input from and the output to the same network connection.
-The fount represents data coming in over the connection, and the drain represents data going back out over that same connection.
+In the above example, ``echo`` requires a :api:`tubes.listening.Flow <flow>` as an argument.
+A ``Flow`` represents the other end of the connection that we just received, both the data we are receiving from that connection and the ability to send data to it.
```
Comment (wsanchez) [inline docs/tube.rst]: I think you can say this more directly:

> A Flow represents the connection that we just received: a stream of inbound data, which we call a fount, and a stream of outbound data, which we call a drain.

File: docs/tube.rst [hunk 8]
```diff
@@ -28,19 +28,29 @@ On a network, that means an echo server as described in :rfc:`862`.
...
+As such, it has 2 attributes: ``.fount``, which is a :api:`tubes.itube.IFount <fount>`, or a source of data, and ``.drain``, which is a :api:`tubes.itube.IDrain <drain>` , or a place where data eventually goes.
+This object is called a "flow", because it establishes a flow of data from one place to another.
```
Comment (wsanchez) [inline docs/tube.rst]: Nice bit of clarity.  Maybe s/to/to and from/.

**Pull request merged**